### PR TITLE
KTL-1189: add e2e basic test

### PR DESCRIPTION
**Pull request by zoobestik**
### Features:
 * E2E tests:
   * ✅ Default simple single playground render
   * ✅ Run default code
   * ✅ Loader and close button click
   * ✅ Edit and run new code
   * ✅ JS API: init
   * 🔽 crosslink:
     * ✅ default action
     * ✅ hide link for incompatible mode
 * ✅ Add test github CI: _chrome_, _firefox_, _safari_.

**Blocker:** 🛑 it waits #185

### CODE REVIEW

File: tests/crosslink.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  test('hidden for crosslink=disabled', ({ page }) =>
+    checkLinkIsHidden(
+      page,
+      `<code data-crosslink="disabled">Hello, World!</code>`,
+    ));
+
+  test('no link for js-libs', ({ page }) => {
+    test.fixme(
+      true,
+      "Test doesn't work, BUG in code! jsLibs.length -> jsLibs.size",
```
Comment (zoobestik) [inline tests/crosslink.e2e.ts:57]: Lol! I found a real bug! 🫣

There is [**no** `.length`](https://github.com/JetBrains/kotlin-playground/blob/d4d4d1bc6f5ffd39d980430413b11a50fa1f512b/src/executable-code/index.js#L127) for [Set](https://github.com/JetBrains/kotlin-playground/blob/d4d4d1bc6f5ffd39d980430413b11a50fa1f512b/src/executable-code/index.js#L217)!

File: playwright.config.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    command: 'npm run test:server',
+    port: 8000,
+    reuseExistingServer: isDevMode,
+  },
+
+  use: {
+    testIdAttribute: 'data-test',
+    headless: ((value) => (value ? value === 'true' : !isDevMode))(
+      env.TEST_HEADLESS_MODE,
+    ),
```
Comment (nikpachoo) [inline playwright.config.ts:46]: That looks complicated.  Is it really supposed to be like this?

File: tests/utlis/server/playground.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    .map(([key, val]) => `data-${key}="${val}"`)
+    .join(' ');
+}
+
+export async function gotoHtmlWidget(page: Page, html: string): Promise<Page>;
+export async function gotoHtmlWidget(
+  page: Page,
+  config: ScriptOptions,
+  html: string,
+): Promise<Page>;
```
Comment (nikpachoo) [inline tests/utlis/server/playground.ts:19]: Are they duplicates?

**Review by nikpachoo — APPROVED** [review]

**Review by nikpachoo — COMMENTED** [review]
It would be great to add a note in the README on how to run the tests.

File: tests/utlis/server/playground.ts (hunk 3 above)
Comment (zoobestik) [reply tests/utlis/server/playground.ts:19]: 🤔 no. There is overload - with two or three parameters. Or what do you mean?

Comment (nikpachoo) [reply tests/utlis/server/playground.ts:19]: I'm trying to understand why not just mark the configuration as optional param and have one function. 
Does it seem like overloading is type-safer? 
Then, it is good move, no questions.

**Pull request merged**